pypit/scripts/tc_1dspec.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
+
+import sys
+import numpy as np
+from pypit import arutils
+from pypit import arload
+from linetools.spectra.xspectrum1d import XSpectrum1D
+from pkg_resources import resource_filename
+#from pdb as debugger
+
+#msgs = pyputils.get_dummy_logger()
+
+def parser(options=None):
+    import argparse
+
+    description = (
+                  'Script to telluric correct a spec!D file. '
+                  'Currently only works for LRIS 800/10000 grating.'
+                  )
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument("infile", type=str, help="Input file (fits)")
+    parser.add_argument("atm_tran", type=str, help="Atmospheric transmission spectrum")
+    #parser.add_argument("--debug", default=False, i
+    #                    action='store_true', help="Turn debugging on")
+
+    if options is None:
+        args = parser.parse_args()
+    else:
+        args = parser.parse_args(options)
+    return args
+
+# This function needs to be replaced
+def get_lrisr_resolution(wavelength):
+    """
+    This comes from P. van Dokkum's fit
+    to the sky lines
+    """
+
+    l9   = (wavelength - 9000.)/1000.
+    resr = 1.26-0.128*l9+0.168*l9**2+0.1173*l9**3
+    resr = resr/wavelength*3.E5 # km s^-1
+
+    return np.array(resr)
+
+def get_transmission(atm_file, data):
+    """
+    This transmission spectrum comes from the ESO SkyCalc,
+    with default inputs.
+    """
+
+    fname = resource_filename('pypit', 'data/extinction/{}'.format(atm_file))
+    tmp = np.loadtxt(fname)
+
+    # Convert transmission spectrum to angstroms
+    # and clip it to wavelength range of data
+    i = ((tmp[:,0]*10 >= min(data.wavelength.value)) &
+         (tmp[:,0]*10 <= max(data.wavelength.value)))
+    trans = XSpectrum1D.from_tuple((tmp[:,0][i]*10, tmp[:,1][i]))
+
+    # Convolve the atm. transsissiom spectrum to LRIS_R resolution
+    res         = get_lrisr_resolution(trans.wavelength.value)
+    fwhm_pix    = arutils.get_fwhm_pix(trans, inval=20000., outval=res)
+
+    smooth_spec = np.zeros((len(trans.wavelength.value)))
+    for i, fp in enumerate(fwhm_pix):
+        tmp = trans.gauss_smooth(fp)
+        smooth_spec[i] = tmp.flux.value[i]
+
+    # Get model transmission on same wavelength grid
+    # as data
+    model = np.zeros((len(data.wavelength.value), 2))
+    model[:,0] = data.wavelength.value
+    model[:,1] = np.interp(data.wavelength.value,
+                           trans.wavelength.value, smooth_spec)
+
+    return model
+
+def get_fscale(data, tran):
+    from numpy.polynomial.chebyshev import chebfit, chebval
+    from scipy.optimize import minimize
+
+    i = ((data.wavelength.value >= 9250) & (data.wavelength.value <= 9650))
```

Comment:
Let's have 9250 and 9650 be default keywords (that can be changed
on execution).